src/test/java/com/google/sps/EditServletTest.java

```diff
@@ -0,0 +1,520 @@
+  private static final String DESTINATION_NAME = "California";
+  private static final String IMAGE_SRC =
+    "https://lh3.googleusercontent.com/p/AF1QipM7tbCZOj_5SOft9cYgI7un3bmieieqvdYkCPT5=s1600-w400";
+  private static final String TRIP_START_DATE = "2020-08-22";
+  private static final String TRIP_END_DATE = "2020-08-23";
+
+  // Constants to represent the different Event Attributes.
+  private static final String SPACE_NEEDLE = "Space Needle";
+  private static final String SPACE_NEEDLE_ADDRESS = 
+    "Space Needle, Broad Street, Seattle, WA, USA";
+  private static final LocalDateTime SPACE_NEEDLE_START_TIME = 
+      LocalDateTime.of(LocalDate.parse("2020-08-22"), LocalTime.of(11, 30));
+  private static final String SPACE_NEEDLE_PLACE_ID = "1234";
+  private static final String WOODLAND = "Woodland Park Zoo";
+  private static final String WOODLAND_ADDRESS = 
+    "Woodland Park Zoo, Phinney Avenue North, Seattle, WA, USA";
+  private static final LocalDateTime WOODLAND_START_TIME = 
+      LocalDateTime.of(LocalDate.parse("2020-08-22"), LocalTime.of(10, 00));
+      private static final String WOODLAND_PLACE_ID = "5678";
+  private static final int HALF_HOUR = 30;
+
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  @Before
+  public void initEditServlet() {
+    editServlet = new EditServlet();
+  }
+
+  @Before
+  public void setUp() {
+    helper.setUp();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void initTest() {
+    Assert.assertTrue(true);
+  }
+
+  /**
+   * HELPER METHODS FOR TESTING.
+   */
+
+  /**
+   * Helper method to create a User Entity and add it to Datastore.
+   *
+   * @param datastore The Datastore object that places the User in Datastore.
+   * @param email The user email.
+   * @return The User Entity that was placed in Datastore.
+   */
+  public Entity createUserEntity(DatastoreService datastore, String email) {
+
+    Entity userEntity = new Entity(User.USER);
+    userEntity.setProperty(User.USER_EMAIL, email);
+    datastore.put(userEntity);
+    return userEntity;
+  }
+
+  /**
+   * Helper method to create a Trip Entity under the provided User Entity and
+   * add it to Datastore.
+   *
+   * @param datastore The Datastore object that places the Trip in Datastore.
+   * @param userEntity The User Entity that the Trip Entity will be stored under.
+   * @param tripName The name of the trip.
+   * @param destinationName The name of the trip destination.
+   * @param imageSrc The String URL of the image that represents the trip.
+   * @param tripStartDate The String start date of the trip. Must be in 
+      "yyyy-mm-dd" format.
+   * @param tripEndDate The String start end of the trip. Must be in 
+      "yyyy-mm-dd" format.
+   * @return The Trip Entity that was placed in Datastore.
+   */
+  public Entity createTripEntity(DatastoreService datastore, Entity userEntity, 
```

Comment:
I'd actually be fine with it taking just the datastore and userEntity as parameters and hard-coding all other properties of the trip. That would be poor form outside the context of a test, but here it might be fine because 1) this method is always called with the same arguments, and 2) there are no assertions about the properties of the trip. In essence this method is used to _put some trip in the datastore_ but we never care about the exact properties of the trip which is put there. But like Bootsie said this morning, this is more a matter of preference than strict style guidance. If you feel that having the parameters makes the tests more readable, then there's no need to change. I think there's good arguments either way.

Reply:
Yeah, that makes sense. I'm going to leave this currently as-is because future tests could require building of two different Trip objects (such as the current tests in this CL with two TripDay or two Event objects), and because I'd prefer to maintain uniform design decisions throughout the helper methods (the `createTripDayEntity(...)` method requires taking parameters). Thanks for this note 👍